packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1466,11 +1478,126 @@ export class ContainerRuntime extends EventEmitter
                 builder.addWithStats(key, contextSummary);
             }));
 
+        const deletedDataStores = this.runGarbageCollection();
+        builder.addBlob(".deletedDataStores", JSON.stringify(deletedDataStores));
+
         this.serializeContainerBlobs(builder);
         const summary = builder.getSummaryTree();
         return { ...summary, id: "" };
     }
 
+    public runGarbageCollection(): string[] {
+        // Maintains a map of dataStoreId to an IReferences object. The IReferences object keeps track of whether
+        // the data store has been visited (to detect loops) and the fluid objects in it that are referenced.
+        const dataStoreReferencesMap: Map<string, IReferences> = new Map();
```

Comment:
FYI: This should be a path in the future, i.e. string[]. to account for nested data stores. string might be fine for now, but if we can keep things generic with clear comments on why, then it would be better.

Reply:
Okay, I would love to understand about nested data stores and what the timeline for that is. I think it make it easier for me to implement keeping that in mind.